# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JitActivation.cpp
# Commit: b620533c6d63
# Full Hash: b620533c6d633ef50c5e5cb3ecd0f398206eb271
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-17 21:52:06
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r f7ec4a920540 -r b620533c6d63 js/src/vm/JitActivation.cpp
--- a/js/src/vm/JitActivation.cpp	Wed Jun 17 11:41:13 2020 +0000
+++ b/js/src/vm/JitActivation.cpp	Wed Jun 17 12:04:54 2020 +0000
@@ -231,9 +231,9 @@
   MOZ_ASSERT(unwound == (trap == wasm::Trap::IndirectCallBadSig));
 
   void* pc = unwindState.pc;
-  wasm::Frame* fp = unwindState.fp;
+  const wasm::Frame* fp = wasm::Frame::fromUntaggedWasmExitFP(unwindState.fp);
 
-  const wasm::Code& code = fp->tls->instance->code();
+  const wasm::Code& code = fp->instance()->code();
   MOZ_RELEASE_ASSERT(&code == wasm::LookupCode(pc));
 
   // If the frame was unwound, the bytecodeOffset must be recovered from the